apps/api-documenter/src/documenters/YamlDocumenter.ts

```diff
@@ -666,7 +666,7 @@ export class YamlDocumenter {
       onGetFilenameForApiItem: (apiItem: ApiItem) => {
         // NOTE: GitHub's markdown renderer does not resolve relative hyperlinks correctly
         // unless they start with "./" or "../".
-        return `xref:${this._getUid(apiItem)}`;
+        return `xref:${this._getUid(apiItem).replace(/[#?]/g, s => encodeURIComponent(s))}`;
```

Comment:
@rbuckton Why are you encoding ONLY the `#` character?  If the xref target is a URL, shouldn't the entire string be URL encoded?

For example, suppose my declaration reference is `Statistics#"total%"`.  Seems like maybe that should get encoded as `Statistics%23%22total%25%22`.

Reply:
That's a fair point. I'll update the commit.

Reply:
Might be useful to consult the DocFX specs to see if they provide any guidance about this.  Or at least try some weird strings to confirm that they are decoding how we expect.  (Encoding/decoding is something that developers almost never get right.)